### PR TITLE
Allow roles to add manga

### PR DESCRIPTION
**Pull request by ethan-sonza**

### CODE REVIEW

File: cogs/manga_main.py [hunk 1]
```diff
@@ -271,7 +271,11 @@ async def add(self, ctx, manga: Option(str, description="The name of the manga s
         else:
             modeval = "user"
             mode = None
+
+        # error embeds
         setupError = discord.Embed(title="Error", color=0xff4f4f, description="Sorry! Please run the setup command first.")
+        permissionError = discord.Embed(title="Error", color=0xff4f4f, description=("You don't have permission to add manga. Allow roles to add manga with `/server add_role`."))
```
Comment (jckli) [inline cogs/manga_main.py:277]: I would rather you move this to line 297, just for organization sake as it isn't used anywhere else I believe.

File: README.md [hunk 2]
```diff
@@ -63,4 +63,16 @@ If you have any issues, please don't be afraid to raise an issue on [GitHub](htt
...
 **manga remove**: Removes a manga from your list.
 
-**manga setgroup**: Sets a manga's scan group. Only that scan group's chapter updates for that manga will be sent.
\ No newline at end of file
+**manga setgroup**: Sets a manga's scan group. Only that scan group's chapter updates for that manga will be sent.
+
+## Environment Variables
+**Discord**
+- `TOKEN`: Discord bot private key
+**Mongo DB**
```
Comment (jckli) [inline README.md:71]:
```suggestion

**Mongo DB**
```

File: README.md [hunk 3]
```diff
@@ -63,4 +63,16 @@ If you have any issues, please don't be afraid to raise an issue on [GitHub](htt
...
+- `MONGO_HOST`: Mongo url link
+- `MONGO_USER`: Mongo username
+- `MONGO_PASS`: Mongo password
+- `MONGO_DB_NAME`: Mongo database name
+**[MangaUpdates](https://www.mangaupdates.com/)**
```
Comment (jckli) [inline README.md:76]:
```suggestion

**[MangaUpdates](https://www.mangaupdates.com/)**
```

File: README.md [hunk 4]
```diff
@@ -63,4 +63,16 @@ If you have any issues, please don't be afraid to raise an issue on [GitHub](htt
 
 **manga remove**: Removes a manga from your list.
 
-**manga setgroup**: Sets a manga's scan group. Only that scan group's chapter updates for that manga will be sent.
\ No newline at end of file
+**manga setgroup**: Sets a manga's scan group. Only that scan group's chapter updates for that manga will be sent.
+
+## Environment Variables
+**Discord**
+- `TOKEN`: Discord bot private key
```
Comment (jckli) [inline README.md:70]:
```suggestion
- `TOKEN`: Discord Bot Token
```

**Review by jckli — CHANGES_REQUESTED** [review]
Other than these changes, it LGTM! I may also change some of the variables just for organization sake as well as test it a bit after these changes, but thanks for the PR!

File: core/mongodb.py [hunk 5]
```diff
@@ -8,11 +8,15 @@
 
 class Mongo:
     def __init__(self):
-        ca = certifi.where()
-        username = os.environ.get("MONGO_USER")
-        password = os.environ.get("MONGO_PASS")
         database_name = os.environ.get("MONGO_DB_NAME")
-        mongo = MongoClient(f"mongodb+srv://{username}:{password}@akane.dsytm.mongodb.net/{database_name}?retryWrites=true&w=majority", tlsCAFile=ca)
+        host = os.environ.get("MONGO_HOST")
```
[the remark this answers is not in the document]
Comment (ethan-sonza) [reply core/mongodb.py:12]: No problem, will revert these. Was mainly a convenience for me to setup a local mongodb host.

File: cogs/manga_main.py [hunk 6]
```diff
@@ -587,7 +592,8 @@ async def testsending(self, ctx):
                         await ctx.respond(embed=noChannelError, view=None)
                         return
                     mangaTestEmbed = discord.Embed(title=f"Testing Update", url="https://picsiv.hayasaka.moe/", description=f"This is a test alert for the MangaUpdates Bot.", color=0x3083e3)
-                    mangaTestEmbed.set_author(name="MangaUpdates", icon_url=self.bot.user.avatar.url)
+                    if self.bot.user and self.bot.user.avatar:
```
[the remark this answers is not in the document]
Comment (ethan-sonza) [reply cogs/manga_main.py:595]: My bot has no avatar/icon, so I was getting the error: `'None' has no '.url'`

File: cogs/manga_general.py [hunk 7]
```diff
@@ -215,7 +233,26 @@ async def delete(self, ctx):
             await mongo.remove_server(ctx.guild.id)
             completeEmbed = discord.Embed(title="Delete Account", color=0x3083e3, description="Successfully deleted your account.")
             await confirm.interaction.response.edit_message(embed=completeEmbed, view=None)
-        
+
+    @server.command(name="add_role", description="Add roles that are allowed to add to the manga list")
```
[the remark this answers is not in the document]
Comment (ethan-sonza) [reply cogs/manga_general.py:237]: What if we make a separate prefix for this? eg. `/role add @ROLE`

File: cogs/manga_main.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jckli) [reply cogs/manga_main.py:595]: Ah interesting, yeah I didn't think about that. Makes sense.

**Comment by jckli** [comment]
Hmm, testing this, the manga remove command is not changed and still checks for administrator permission. Other than that everything seems working and fine. I will merge this into a branch on the original mangaupdates-bot repository and change some variable names and fix that.

Sorry about not being very active these past couple days. Had a bunch of work to do. :/

**Pull request merged**